File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        int k1=0, k2=0;
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='('){
                k1++;
            }
            else{
                if(k1>0){
                    k1--;
                }
                else{
                    k2++;
                }
            }
        }
        return k1+k2;
    }
}

File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        k1=0
        k2=0
        for c in range(len(s)): 
            if s[c]=='(':
                k1+=1
            else:
                if k1>0:
                    k1-=1
                else:
                    k2+=1
                   
        return k1+k2;